Load list of tags to skip from file

msherman64 opened the pull request:
kolla-ansible supports the --skip-tags option

This adds a simple list of tags to a file. Uncomment lines to skip.

msherman64 committed:
Load list of tags to skip from file

kolla-ansible supports the --skip-tags option

This adds a simple list of tags to a file. Uncomment lines to skip.

diurnalist commented on kolla-skip-tags:
```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
```
Doesn't need this does it?

msherman64 replied:
This was just to enable syntax highlighting in general. It's getting evaluated by bash / gnu functions, so it seemed reasonable to include.

diurnalist commented on cc-ansible:
```diff
@@ -325,4 +325,11 @@ if [[ "${POSARGS[*]}" =~ post-deploy ]]; then
 
+#load tags to skip
+ANSIBLE_SKIP_TAGS="$(grep -v "^#" ./kolla-skip-tags | paste -sd ',')"
+if [[ "${ANSIBLE_SKIP_TAGS}" ]]; then
```
Huh, I didn't know it could be that simple. Is there a difference b/w this and `-n` ?

msherman64 replied:
With no arguments, `[[ expr ]]` tests for "is null", where `-n` tests for "is length nonzero"
From what I can find: https://unix.stackexchange.com/a/280432 they result in the same behavior, but `-n` is semantically clearer.

diurnalist commented on cc-ansible:
```diff
@@ -325,4 +325,11 @@ if [[ "${POSARGS[*]}" =~ post-deploy ]]; then
+ANSIBLE_SKIP_TAGS="$(grep -v "^#" ./kolla-skip-tags | paste -sd ',')"
+if [[ "${ANSIBLE_SKIP_TAGS}" ]]; then
+  kolla_args+=("--skip-tags")
+  kolla_args+=("${ANSIBLE_SKIP_TAGS[@]}")
```
Do you need the `[@]`?

msherman64 replied:
I need to check, that's left over from when it was getting interpreted as an array.

diurnalist reviewed (approved):
Looks good, I like that they are in order!

msherman64 commented:
The list of tags was generated by running the raw ansible-playbook command with `--list-tags`, then manually parsing it.
It might be good to have some kind of CI process generate that list, so it stays up to date.

msherman64 committed:
Clean up shell syntax

[[ expr ]] and [[ -n expr ]] evaluate the same, but -n is clearer.
Is null, vs is nonzero length. In bash, a zero length string is null.
